src/server/snapshot.h

```diff
@@ -49,8 +49,18 @@ struct Entry;
 // over the sink until explicitly stopped.
 class SliceSnapshot {
  public:
-  SliceSnapshot(DbSlice* slice, CompressionMode compression_mode,
-                std::function<void(std::string)> on_push, std::function<void()> on_snapshot_finish);
+  // Represents a target for receiving snapshot data.
+  struct SnapshotDataConsumer {
```

Comment:
Consider calling it `SnapshotDataConsumerInterface` to emphasize this is an interface